src/test/kotlin/com/terraformation/backend/db/TimeZonePopulatorTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.terraformation.backend.db
+
+import com.terraformation.backend.db.default_schema.UserId
+import com.terraformation.backend.db.default_schema.tables.pojos.TimeZonesRow
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+
+internal class TimeZonePopulatorTest : DatabaseTest() {
+  private val populator by lazy { TimeZonePopulator(dslContext) }
+
+  private val userId = UserId(1000)
+
+  private val invalidTimeZone = "Bogus/Nowhere"
+  private val validTimeZone = "America/New_York"
+
+  @Test
+  fun `inserts time zones that are not already in the database`() {
+    // A time zone that is extremely unlikely to be removed from the tz database
+    populator.populateTimeZonesTable()
+
+    assertEquals(
+        TimeZonesRow(validTimeZone),
+        timeZonesDao.fetchOneByTimeZone(validTimeZone),
+        "Time zone should have been inserted")
+  }
+
+  @Test
+  fun `deletes time zones that are no longer in the Java time zones list`() {
+    timeZonesDao.insert(TimeZonesRow(invalidTimeZone))
+
+    insertUser(userId, timeZone = invalidTimeZone)
```

Comment:
just curious why postgres doesn't complain on this.

Reply:
Which part? You mean inserting the invalid time zone? It's because from Postgres's point of view, the time zone is just a string value with no particular meaning.

Reply:
Thanks for the clarification. Yes, makes sense now.